src/log/log_mgr.c

```diff
@@ -656,8 +656,25 @@ __log_file_server(void *arg)
                  * make sure the compiler does not reorder the following two statements.
                  */
                 WT_ASSIGN_LSN(&close_end_lsn, &log->log_close_lsn);
+
+                /*
+                 * Based on the code investigation, there is no reason to use a FULL_BARRIER here.
+                 * We send a message that `log_close_lsn` is read by setting `log_close_fh` to NULL,
+                 * making it available to `__log_newfile` to assign a new one. Therefore, release
+                 * semantics should be sufficient.
+                 *
+                 * For now, this is placed under the TSAN_BUILD macro only, since weakening barriers
+                 * requires extra care.
+                 *
+                 * FIXME-WT-15337: Consider using a release write in production.
```

Comment:
I take a different view to you here. If TSAN considers the new code to be kosher, and we're confident in our log subsystem testing, I think we can be a little bolder and make this change for all configurations.

(In fact, even if we're not confident in our log subsystem testing, I don't think the fix is to hide changes behind `ifdef`s -- the more we hammer on it, the better.)

Reply:
Thanks. I wasn't aware of the context, please don't let this comment thread hold up the PR.